packages/dd-trace/src/lambda/runtime/patch.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const path = require('path')
+
+const { _extractModuleNameAndHandlerPath, _extractModuleRootAndHandler, _getLambdaFilePath } = require('./ritm')
+const { datadog } = require('../handler')
+const { addHook } = require('../../../../datadog-instrumentations/src/helpers/instrument')
+const shimmer = require('../../../../datadog-shimmer')
+
+/**
+ * Patches a Datadog Lambda module by calling `patchDatadogLambdaHandler`
+ * with the handler name `datadog`.
+ *
+ * @param {*} datadogLambdaModule node module to be patched.
+ * @returns a Datadog Lambda module with the `datadog` function from
+ * `datadog-lambda-js` patched.
+ */
+const patchDatadogLambdaModule = (datadogLambdaModule) => {
+  shimmer.wrap(datadogLambdaModule, 'datadog', patchDatadogLambdaHandler)
+
+  return datadogLambdaModule
+}
+
+/**
+ * Patches a Datadog Lambda handler in order to do
+ * Datadog instrumentation by getting the Lambda handler from its
+ * arguments.
+ *
+ * @param {*} datadogHandler the Datadog Lambda handler to destructure.
+ * @returns the datadogHandler with its arguments patched.
+ */
+function patchDatadogLambdaHandler (datadogHandler) {
+  return (userHandler) => {
+    return datadogHandler(datadog(userHandler))
+  }
+}
+
+/**
+ * Patches a Lambda module on the given handler path.
+ *
+ * @param {string} handlerPath path of the handler to be patched.
+ * @returns a module with the given handler path patched.
+ */
+const patchLambdaModule = (handlerPath) => (lambdaModule) => {
+  shimmer.wrap(lambdaModule, handlerPath, patchLambdaHandler)
+
+  return lambdaModule
+}
+
+/**
+ * Patches a Lambda handler in order to do Datadog instrumentation.
+ *
+ * @param {*} lambdaHandler the Lambda handler to be patched.
+ * @returns a function which patches the given Lambda handler.
+ */
+function patchLambdaHandler (lambdaHandler) {
+  return datadog(lambdaHandler)
+}
+
+/**
+ * Immediately Invoked Function Expression (IIFE) to avoid
+ * error on previously defined constants.
```

Comment:
What's the error? Not sure why this should be necessary?

Reply:
Oh, previously when I started working on this, `patch.js` and `registerLambdaHook()` from `ritm.js` would conflict with the variable naming, stating that both of them were already registered, somehow I tried it again to replicate the error and it's gone, I'll update it!